android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/AllFilter.kt

```diff
@@ -0,0 +1,762 @@
+import java.text.SimpleDateFormat
+import java.time.ZoneOffset
+import java.util.Date
+import java.util.Locale
+
+@SuppressWarnings("CyclomaticComplexMethod","LongMethod","ComplexCondition")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AllFilter(
+	currentFilterVals: HomePageUiState.FiltersModel,
+	updateFilterVals: (HomePageUiState.FiltersModel) -> Unit,
+	coroutineScope: CoroutineScope,
+	closeAction: () -> Unit,
+) {
+	val listState: LazyListState = rememberLazyListState()
+	var housingPref by remember {
+		mutableStateOf(
+			currentFilterVals.housingType
+		)
+	}
+	var startDateButtonText by remember {
+		mutableStateOf(
+			currentFilterVals.dateRange.startingDate?.let {
+				parseUTCDateTimeToLocal(it)
+			} ?: ""
+		)
+
+	}
+	var endDateButtonText by remember {
+		mutableStateOf(
+			currentFilterVals.dateRange.endingDate?.let {
+				parseUTCDateTimeToLocal(it)
+			} ?: ""
+		)
+	}
+	var lowerPriceBoundText by remember {
+		mutableStateOf(
+			currentFilterVals.priceRange.lowerBound?.toString() ?: ""
+		)
+
+	}
+	var upperPriceBoundText by remember {
+		mutableStateOf(
+			currentFilterVals.priceRange.upperBound?.toString() ?: ""
+		)
+	}
+	var lowerLocBound by remember {
+		mutableIntStateOf(
+			currentFilterVals.locationRange.lowerBound ?: 0
+		)
+	}
+
+	var upperLocBound by remember {
+		mutableIntStateOf(
+			currentFilterVals.locationRange.upperBound ?: MAX_LOCATION_RANGE
+		)
+	}
+	var sliderPosition by remember { mutableStateOf(lowerLocBound.toFloat()..upperLocBound.toFloat()) }
+	var lowerLocBoundText by remember {
+		mutableStateOf(
+			currentFilterVals.locationRange.lowerBound?.toString() ?: ""
+		)
+	}
+	var upperLocBoundText by remember {
+		mutableStateOf(
+			currentFilterVals.locationRange.upperBound?.toString() ?: ""
+		)
+	}
+	var isFavourite by remember {
+		mutableStateOf(
+			currentFilterVals.favourite
+		)
+	}
+	var lowerLocTextFieldError by remember { mutableStateOf(false) }
+	var upperLocTextFieldError by remember { mutableStateOf(false) }
+	var lowerPriceTextFieldError by remember { mutableStateOf(false) }
+	var upperPriceTextFieldError by remember { mutableStateOf(false) }
+	val dateRangePickerState = rememberDateRangePickerState()
+	var openDatePicker by rememberSaveable { mutableStateOf(false) }
+	var bedroomForSublet by remember {
+		mutableStateOf(
+			currentFilterVals.roomRange.bedroomForSublet
+		)
+	}
+	var bedroomInProperty by remember {
+		mutableStateOf(
+			currentFilterVals.roomRange.bedroomInProperty
+		)
+	}
+	var bathroom by remember {
+		mutableStateOf(
+			currentFilterVals.roomRange.bathroom
+		)
+	}
+	var ensuitebathroom by remember {
+		mutableStateOf(
+			currentFilterVals.roomRange.ensuiteBathroom
+		)
+	}
+	var genderPref by remember {
+		mutableStateOf(
+			currentFilterVals.gender
+		)
+	}
+	val datePickerBottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+	BasicFilterLayout(
+		modifier = Modifier.height(800.dp),
```

Comment:
Oh, actually, can't you just make this fill height for this one, since it covers the whole screen?